phoenix-core/src/main/java/org/apache/phoenix/schema/transform/Transform.java

```diff
@@ -71,14 +87,150 @@ public static void addTransform(PhoenixConnection connection, String tenantId, P
+        PName newTableNameWithoutSchema = PNameFactory.newName(SchemaUtil.getTableNameFromFullName(systemTransformParams.getNewPhysicalTableName()));
+        PTable newTable = new PTableImpl.Builder()
+                .setTableName(newTableNameWithoutSchema)
+                .setParentTableName(table.getParentTableName())
+                .setBaseTableLogicalName(table.getBaseTableLogicalName())
+                .setPhysicalTableName(newTableNameWithoutSchema)
+                .setAllColumns(table.getColumns())
+                .setAppendOnlySchema(table.isAppendOnlySchema())
+                .setAutoPartitionSeqName(table.getAutoPartitionSeqName())
+                .setBaseColumnCount(table.getBaseColumnCount())
+                .setBucketNum(table.getBucketNum())
+                .setDefaultFamilyName(table.getDefaultFamilyName())
+                .setDisableWAL(table.isWALDisabled())
+                .setEstimatedSize(table.getEstimatedSize())
+                .setFamilies(table.getColumnFamilies())
+                .setImmutableRows(table.isImmutableRows())
+                .setIsChangeDetectionEnabled(table.isChangeDetectionEnabled())
+                .setIndexType(table.getIndexType())
+                .setName(newTableName)
+                .setMultiTenant(table.isMultiTenant())
+                .setParentName(table.getParentName())
+                .setParentSchemaName(table.getParentSchemaName())
+                .setPhoenixTTL(table.getPhoenixTTL())
+                .setNamespaceMapped(table.isNamespaceMapped())
+                .setSchemaName(table.getSchemaName())
+                .setPkColumns(table.getPKColumns())
+                .setPkName(table.getPKName())
+                .setPhoenixTTLHighWaterMark(table.getPhoenixTTLHighWaterMark())
+                .setRowKeySchema(table.getRowKeySchema())
+                .setStoreNulls(table.getStoreNulls())
+                .setTenantId(table.getTenantId())
+                .setType(table.getType())
+                // SchemaExtractor uses physical name to get the table descriptor from. So we use the existing table here
+                .setPhysicalNames(ImmutableList.copyOf(table.getPhysicalNames()))
+                .setUpdateCacheFrequency(table.getUpdateCacheFrequency())
+                .setTransactionProvider(table.getTransactionProvider())
+                .setUseStatsForParallelization(table.useStatsForParallelization())
+                // TODO SET SCHEMAVERSION
+                // Transformables
+                .setImmutableStorageScheme(
+                        (changedProps.getImmutableStorageSchemeProp() != null? changedProps.getImmutableStorageSchemeProp():table.getImmutableStorageScheme()))
+                .setQualifierEncodingScheme(
+                        (changedProps.getColumnEncodedBytesProp() != null? changedProps.getColumnEncodedBytesProp() : table.getEncodingScheme()))
+                .build();
```

Comment:
One thing to handle in a future JIRA is how this will co-exist with schema registry -- when we transform we probably need to create a new schema entry in the external schema registry because change detection will need to know the column encoding to be able to parse WAL edits.

Reply:
Noted